dsl/src/main/groovy/org/cristalise/dsl/persistency/outcome/OutOfSpecs.groovy

```diff
@@ -28,5 +28,18 @@ import groovy.transform.CompileStatic;
 @CompileStatic
 class OutOfSpecs {
     String pattern = null
+    String expression = null
     String message = null
+
+    public void setPattern(String p) {
+        if (pattern && expression) throw new InvalidDataException("Do not use pattern and expression at the same time")
```

Comment:
Why not? Both can raise a warning.

Reply:
in that case expression shall have a different message.

Reply:
but, I think it is an overkill to specify both OOS pattern and expression for a single field.